clients/client/src/main/java/org/projectnessie/client/util/v2api/ClientSideDeleteNamespace.java

```diff
@@ -0,0 +1,99 @@
+ * limitations under the License.
+ */
+package org.projectnessie.client.util.v2api;
+
+import java.util.Map;
+import java.util.Optional;
+import org.projectnessie.client.api.NessieApiV2;
+import org.projectnessie.client.builder.BaseDeleteNamespaceBuilder;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNamespaceNotEmptyException;
+import org.projectnessie.error.NessieNamespaceNotFoundException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.error.NessieReferenceNotFoundException;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.EntriesResponse;
+import org.projectnessie.model.Namespace;
+import org.projectnessie.model.Operation;
+
+/**
+ * Supports previous "delete namespace" functionality of the java client over Nessie API v2.
+ *
+ * <p>API v2 does not have methods dedicated to manging namespaces. Namespaces are expected to be
+ * managed as ordinary content objects.
+ */
+public final class ClientSideDeleteNamespace extends BaseDeleteNamespaceBuilder {
+  private final NessieApiV2 api;
+
+  public ClientSideDeleteNamespace(NessieApiV2 api) {
+    this.api = api;
+  }
+
+  @Override
+  public void delete()
+      throws NessieNamespaceNotFoundException, NessieReferenceNotFoundException,
+          NessieNamespaceNotEmptyException {
+    ContentKey key = ContentKey.of(namespace.getElements());
+    Map<ContentKey, Content> contentMap;
+    try {
+      contentMap = api.getContent().refName(refName).hashOnRef(hashOnRef).key(key).get();
+    } catch (NessieNotFoundException e) {
+      throw new NessieReferenceNotFoundException(e.getMessage(), e);
+    }
+
+    Optional<Object> existingNamespace =
+        Optional.ofNullable(contentMap.get(key)).flatMap(c -> c.unwrap(Namespace.class));
+
+    if (!existingNamespace.isPresent()) {
+      throw new NessieNamespaceNotFoundException(
+          String.format("Namespace '%s' does not exist", key.toPathString()));
+    }
+
+    Optional<EntriesResponse.Entry> entry;
+    try {
+      entry =
+          api
+              .getEntries()
+              .refName(refName)
+              .hashOnRef(hashOnRef)
+              .filter(String.format("entry.namespace.startsWith('%s')", key))
+              .stream()
+              .findAny();
+    } catch (NessieNotFoundException e) {
+      throw new NessieReferenceNotFoundException(e.getMessage(), e);
+    }
+
+    if (entry.isPresent()) {
+      throw new NessieNamespaceNotEmptyException(
+          String.format("Namespace '%s' is not empty", key.toPathString()));
+    }
+
+    try {
+      api.commitMultipleOperations()
+          .branchName(refName)
+          .hash(hashOnRef)
+          .commitMeta(CommitMeta.fromMessage("delete namespace " + key))
+          .operation(Operation.Delete.of(key))
+          .commit();
```

Comment:
Would be nice to return the result of the commit() (follow-up)

Reply:
That would change the java client method signature and break existing clients. We can add a new method to `UpdateNamespaceBuilder` for that, though.

Reply:
Maybe under #5487 ?